plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -66,9 +66,11 @@ def set_deployment_metadata
       end
     end
 
+    # labels that are used to match the previous deployment,
+    # they cannot change or we will create a new deployment instead of updating the previous one
+    # re-creating roles or re-creating the deploy group will lead to duplicate deployments
     def deployment_labels
-      # Deployment labels should not include the ids of the release, role or deploy groups
-      release_doc_metadata.except(:release_id, :role_id, :deploy_group_id)
+      release_doc_metadata.slice(:project_id, :role_id, :deploy_group_id)
```

Comment:
just tried a few deploys without any label restrictions and it won't take down the pods when deleting the deploy ... so we create duplicate pods and they never get cleaned up

with this .slice implementation the validation failed:

```
JobExecution failed: Deployment.extensions "truth-service-app-server" is invalid: spec.template.metadata.labels: Invalid value: {"deploy_group":"production","deploy_group_id":"1","deploy_id":"111","project":"example-project","project_id":"1","release_id":"79","revision":"617182f02ecdf2b38a46978e00d2ff755811c098","role":"app-server","role_id":"6","tag":"617182f02ecdf2b38a46978e00d2ff755811c098"}: `selector` does not match template `labels
```

since the project name in samson and the kubernetes file did not match ... good times

so the simple solution is to for now only put role and project in there ... since that is the only thing we need to be unique ... and for future PR stop messing with the labels altogether ...